grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -402,9 +402,16 @@ class OrderController {
                     if (orderAdjustment.hasRegularInvoice) {
                         throw new UnsupportedOperationException("${warehouse.message(code: 'errors.noPermissions.label')}")
                     }
-                    orderInstance.removeFromOrderAdjustments(orderAdjustment)
-                    orderAdjustment.delete()
-                    if (!orderInstance.hasErrors() && orderInstance.save(flush: true)) {
+
+                    try {
+                        orderInstance.removeFromOrderAdjustments(orderAdjustment)
+                        orderAdjustment.delete()
+                        orderInstance.save(flush: true)
+                    } catch(Exception e) {
```

Comment:
I am not sure if catching 'vague' exception and throw `UnsupportedOperationException` is a good approach. What is the exception that you are getting here?

Reply:
yes, you are probably right
the exception that I am getting is 
```
Cannot delete or update a parent row: a foreign key constraint fails ('openboxes'.'order_adjustment_invoice', CONSTRAINT 'order_adjustment_invoice_ibfk_1' FOREIGN KEY ('order_adjustment_id') REFERENCES 'order_adjustment' ('id')
```

and so I am not sure how to deal with that then.

Reply:
On second thought your try-catch approach is definitely a wrong approach. You should check if the item or adjustment has an invoice associated and then throw `UnsupportedOperationException`.